libvast/vast/chunk.hpp

```diff
@@ -166,4 +179,13 @@ class chunk : public caf::ref_counted {
   deleter_type deleter_;
 };
 
+// We need this template in order to be able to send chunks as messages,
+// but we never actually want to have to use it since it will only be invoked
+// if chunk messages would be sent over the network. And we only want to send
+// chunks around internally. So it is declared here but never defined, leading
+// to a linker error if its used.
```

Comment:
Intriguing idea, I like it.

What's the issue with sending chunks over the network, though? Can't we materialize them as an owning byte buffer on the receiver side, using the default allocator?

Reply:
We can, but we currently dont need to do that. So I think its better to wait until we have a use case for that and then enable it explicitly.

Reply:
I agree, let's hold off until we have that use case.